app/controllers/posts_controller.rb

```diff
@@ -0,0 +1,29 @@
+class PostsController < ApplicationController
+  before_action :require_log_in, only: [:new, :create]
+
+  def index
+    @posts = Post.all
+  end
+
+  def new
+    @post = Post.new
+  end
+
+  def create
+    @post = Post.new(whitelist_params)
+    @post.user_id = current_user.id
```

Comment:
These lines can be combined: `@post = current_user.posts.build(whitelist_params)`